docparse/jsonschema.go

```diff
@@ -401,6 +429,62 @@ start:
 	return &p, nil
 }
 
+func dropTypePointers(typ ast.Expr) ast.Expr {
+	var t *ast.StarExpr
+	var ok bool
+	for t, ok = typ.(*ast.StarExpr); ok; t, ok = typ.(*ast.StarExpr) {
+		typ = t.X
+	}
+	return typ
+}
+
+func findTypeIdent(typ ast.Expr, curPkg string) (*ast.Ident, string, error) {
+	typ = dropTypePointers(typ)
+	if i, ok := typ.(*ast.Ident); ok {
+		// after droping the stars we have the ident:
+		return i, curPkg, nil
+	}
+
+	se, ok := typ.(*ast.SelectorExpr)
+	if !ok {
+		// not ident, not a package selector expr, cannot find ident
+		return nil, "", fmt.Errorf("fieldTypeIdent: cannot find ident for type: %T", typ)
+	}
+
+	pkgSel, ok := se.X.(*ast.Ident)
+	if !ok {
+		return nil, "", fmt.Errorf("fieldTypeIdent: SelectorExpr's typ.X is not ast.Ident: %#v", se.X)
+	}
+	return se.Sel, pkgSel.Name, nil
+}
+
+func lookupTypeAndRef(file, pkg, name string) (string, string, error) {
+	// Check if the type resolves to a Go primitive.
+	lookup := pkg + "." + name
+	ts, _, _, err := findType(file, pkg, name)
+	if err != nil {
+		return "", "", err
+	}
+	/*
```

Comment:
would be cool to leave these outside of master I suppose, but other than that I guess that's good to go